### DISCUSSION TITLE
Related fields not saving

### DISCUSSION THREAD

**Discussion by infiniteroles** [Laravel-Backpack/community-forum#1130] [Q&A (Help)]
Hello, I’m having an issue with the related fields in my CRUD. I have a CRUD that displays several fields from its own model, and two fields from a related model. I’m doing it like this:
`CRUD::addField([
    'name' => 'risk',
    'label' => 'Risk',
    'type' => 'select',
    'entity' => 'user',
    'attribute' => 'risk',
    'allows_null' => false,
    'model' => "App\Models\User",
]);`

And in the model, I have:
`public function user()
{
    return $this->belongsTo(User::class);
}`
When I modify any field and save, those changes are saved correctly, but if I modify the “risk” field, no changes are saved in the database, and I don’t get any errors either.

Thank you very much,
Oscar

**Comment by karandatwani92** [comment]
Hey @infiniteroles 

Try this:
```
[
    'name' => 'risk_id', // the DB column for the foreign key, ensure it's fillable.
    'type' => 'select',
    'model'     => "App\User", // related model
    'attribute' => 'name', // attribute to display on select box
    'allow_null' => false,
],
```
OR

```
[
    'name' => 'risk_id', // the DB column for the foreign key, ensure it's fillable.
    'type' => 'select',
    'entity' => 'risk', // while using entity, column name should be `relationship name` + `_id`; example: risk_id
    'allow_null' => false,
]
```

**Reply by infiniteroles** [reply]
thanks karandatwani92 but IO dont have a risk_id, I have this two models:
AccountVerification:

`<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Carbon;

class AccountVerification extends Model
{
    use CrudTrait;
    use HasFactory;

    /**
     * TABLE NAME
     *
     */
    public $table = "account_verifications";

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'user_id',
        'date',
        'status',
        'type',
        //
        // Person+Company
        'name',
        'surname',
        'surname2',
        ...

    ];


    public function user()
    {
        return $this->belongsTo(User::class)->withTrashed();
    }

    public function getUserEmailAttribute()
    {
        return $this->user->email;
    }

    public function getUserNameAttribute()
    {
        return $this->user->name;
    }





}`

And USER:
`<?php

namespace App\Models;

use Backpack\CRUD\app\Models\Traits\CrudTrait;
use App\Observers\UserObserver;
use Illuminate\Contracts\Auth\MustVerifyEmail;
use Illuminate\Database\Eloquent\Builder;


class User extends Authenticatable implements MustVerifyEmail
{
    use CrudTrait;
    //use HasApiTokens;
    use HasFactory;
    use HasProfilePhoto;

    // Borrado usuarios
    use SoftDeletes;

    // Admin Role ID number
    public $adminID = 100;

    /**
     * The attributes that are mass assignable.
     *
     * @var array
     */
    protected $fillable = [
        'id',
        'name',
        'password',
        'user_type_id',
        'fee_setup_cherry',
        'fee_payment_cherry',
        'fee_setup_user',
        'fee_payment_user',
        'contact_permission_level',
        'risk',
        'risk_date',
        ....
}`

And I want to sabe irsk value in user que it´s changed in accoutnverificationcrud view.

thnaks so much

**Comment by infiniteroles** [comment]
I will open a new one with new information a better data about my issue. Thanks a lot!
